### ISSUE TITLE
[BUG] Anomaly detection features count locked at 5

### ISSUE THREAD

**Issue by MCdeamon7** [opensearch-project/anomaly-detection-dashboards-plugin#1016]
**ISSUE**
The documentation say that plugins.anomaly_detection.max_anomaly_features can be increased up to 10, (and is limiting but better than nothing)
I increased it to 10 via DevTools:
PUT /_cluster/settings
{
  "persistent" : {
    "plugins": {
      "anomaly_detection": {
        "max_anomaly_features": "10"
      }
    }
  }
}
And GET /_cluster/settings/ tell me that is actually persistent even after shutting down the docker compose and bringing it back up.
But at the moment I try to insert the feature in the anomaly detection job in the dashboard, the maximum is locked at 5.

**To Reproduce**
1. Go to DevTools in dashboard
2. send and past the previous command
4. go to anomaly detection
5.try to set more than 5 features.

**Host/Environment (please complete the following information):**
 - OS: [Linux Manjaro]
 - Deployment: Docker compose
 - Version [latest]

**Additional context**
Sorry for my broken English, I'm Italian but trying to improve.

**Comment by andrross** [comment]
@quangdutran Are you still working on this? I've assigned the issue to you, but if you're not working on it I'll remove you as the assignee. Thanks!

**Comment by quangdutran** [comment]
Yes @andrross I am still working on it but kinda struggle a bit with how to call the cluster setting API from BE. Is there any existing pattern to call the API or existing cluster setting variable that I can make use of?

**Comment by quangdutran** [comment]
Hi @amitgalitz I know that we are hard code the value [here](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/eeaedc3f709964ea6a4d436efbbead5af1ac0b9a/public/utils/constants.ts#L82). 

So my first idea is to call the API `/_cluster/settings` from BE to get the maximum features, then save to the constant. But this won't reflect when the limit is changed again. So better to call the API where-ever this constant is being used.

For example in [Features](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/eeaedc3f709964ea6a4d436efbbead5af1ac0b9a/public/pages/ConfigureModel/components/Features/Features.tsx#L91), I dont really know how to call the API from here. [ClusterInfo](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/eeaedc3f709964ea6a4d436efbbead5af1ac0b9a/server/models/types.ts#L20) seems to be something useful but it does not contain that much information about feature limit.

If there is already a piece of code that do similar getting cluster setting, would you mind advice me pls

**Comment by amitgalitz** [comment]
> Hi [@amitgalitz](https://github.com/amitgalitz) I know that we are hard code the value [here](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/eeaedc3f709964ea6a4d436efbbead5af1ac0b9a/public/utils/constants.ts#L82).
> 
> So my first idea is to call the API `/_cluster/settings` from BE to get the maximum features, then save to the constant. But this won't reflect when the limit is changed again. So better to call the API where-ever this constant is being used.
> 
> For example in [Features](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/eeaedc3f709964ea6a4d436efbbead5af1ac0b9a/public/pages/ConfigureModel/components/Features/Features.tsx#L91), I dont really know how to call the API from here. [ClusterInfo](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/eeaedc3f709964ea6a4d436efbbead5af1ac0b9a/server/models/types.ts#L20) seems to be something useful but it does not contain that much information about feature limit.
> 
> If there is already a piece of code that do similar getting cluster setting, would you mind advice me pls

So we don't have code specifically calling cluster settings in the AD Dashboards repo but there are similar examples in other opensearch dashboards plugins like alerting: https://github.com/opensearch-project/alerting-dashboards-plugin/blob/217bf4ddb31c786dd0c25f4e66d88fa7e63802e4/public/pages/utils/helpers.js#L103

What we need to do is make an api call to the cluster settings like we do other API calls in AD, we have some examples of calling opensearch APIs like cluster health or remote info https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/server/routes/opensearch.ts#L556

You can follow the pattern of other API calls and add a cluster setting call while also referencing how its done in alerting, we then want to call this method and use the number of features from the setting instead of a hard coded value.

**Comment by quangdutran** [comment]
Hi @amitgalitz could you help me to review the PR? A couple of notes:
- I am not sure how to write UT test for this feature. If there is a sample, appreciate if you could show me.
- The documentation for this is not necessary I guess.
- Manual test seems ok

![Image](https://github.com/user-attachments/assets/c9dccaa7-ff16-4e4c-bf32-ff7b9e9a526d)
